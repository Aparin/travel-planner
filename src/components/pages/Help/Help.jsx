import React from 'react';
import { Link } from 'react-router-dom';
import Button from '@material-ui/core/Button';

const Help = () => (
  <React.Fragment>
    <Link to="/apps/travel-planner/">
      <Button
        variant="outlined"
        color="primary"
        style={{ marginTop: '10px' }}
      >
      Вернуться
      </Button>
    </Link>
    <h2>Описание функционала и использование</h2>
    <p>
      SPA представляет собой простой планировщик маршрутов.
      Адаптирован только для настольных компьютеров.
    </p>

    <h2>Как добавить точку маршрута</h2>
    <p>1. Выбираем место, это можно сделать 2 способами:</p>
    <p>- ввести адрес, или название объекта в левом верхнем углу карты, или,</p>
    <p>
      - выставить карту так, чтобы нужная точка была примерно в центре карты
      (потом позицию можно будет уточнить).
    </p>
    <p>2. Ввести название точки маршрута в поле слева от карты и нажать &quot;Enter&quot;.</p>

    <h2>Как использовать</h2>
    <p>- Точку маршрута можно удалить нажав на красную корзину,</p>
    <p>- если кликнуть на маркере точки на карте, появится окно с названием точки и адресом,</p>
    <p>
      - маркер на карте можно передвигать, при этом адрес будет
      автоматически менятся (с небольшой задержкой, которую вы скорее всего даже не заметите),
    </p>
    <p>- когда точек маршрута будет больше одной, они начнут соединятся линиями,</p>
    <p>
      - при перетаскивании маркера точки на карте,
      линии соединяющие соседнии точки, также будут перестраиватся,
    </p>
    <p>
      - очерёдность точек маршрута можно менять,
      перетаскивая их имена слева от карты, при этом очерёдность линий,
      соединяющих точки на карте, также будет меняться.
    </p>
  </React.Fragment>
);

export default Help;
