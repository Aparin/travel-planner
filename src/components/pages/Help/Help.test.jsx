/* eslint-disable no-undef */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Help from './Help';

describe('Help', () => {
  it('renders Help without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Help />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('Help snapshot', () => {
    const component = renderer.create(
      <Help />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
