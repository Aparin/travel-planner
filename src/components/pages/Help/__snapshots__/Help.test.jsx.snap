// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help Help snapshot 1`] = `
Array [
  <h2>
    Описание функционала и использование
  </h2>,
  <p>
    SPA представляет собой простой планировщик маршрутов. Адаптирован только для настольных компьютеров.
  </p>,
  <h2>
    Как добавить точку маршрута
  </h2>,
  <p>
    1. Выбираем место, это можно сделать 2 способами:
  </p>,
  <p>
    - ввести адрес, или название объекта в левом верхнем углу карты, или,
  </p>,
  <p>
    - выставить карту так, чтобы нужная точка была примерно в центре карты (потом позицию можно будет уточнить).
  </p>,
  <p>
    2. Ввести название точки маршрута в поле слева от карты и нажать "Enter".
  </p>,
  <h2>
    Как использовать
  </h2>,
  <p>
    - Точку маршрута можно удалить нажав на красную корзину,
  </p>,
  <p>
    - если кликнуть на маркере точки на карте, появится окно с названием точки и адресом,
  </p>,
  <p>
    - маркер на карте можно передвигать, при этом адрес будет автоматически менятся (с небольшой задержкой, которую вы скорее всего даже не заметите),
  </p>,
  <p>
    - когда точек маршрута будет больше одной, они начнут соединятся линиями,
  </p>,
  <p>
    - при перетаскивании маркера точки на карте, линии соединяющие соседнии точки, также будут перестраиватся,
  </p>,
  <p>
    - очерёдность точек маршрута можно менять, перетаскивая их имена слева от карты, при этом очерёдность линий, соединяющих точки на карте, также будет меняться.
  </p>,
  <a
    href="/main"
    onClick={[Function]}
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
      disabled={false}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "marginTop": "10px",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Спланировать путешествие
      </span>
    </button>
  </a>,
]
`;
